src/benchmarks/micro/corefx/System.Globalization/Perf.StringCultureSpecific.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+
+namespace System.Globalization.Tests
+{
+    [BenchmarkCategory(Categories.CoreFX, Categories.CoreCLR)]
+    public class Perf_StringCultureSpecific
+    {
+        private string _value, _same, _diffAtFirstChar, _diffAtLastChar, _firstHalf, _secondHalf;
+
+        public static IEnumerable<(CultureInfo CultureInfo, CompareOptions CompareOptions)> GetCultureOptions()
+        {
+            // Ordinal and OrdinalIgnoreCase use single execution path for all cultures, so we test it only for "en-US"
+            yield return (new CultureInfo("en-US"), CompareOptions.Ordinal);
+            yield return (new CultureInfo("en-US"), CompareOptions.OrdinalIgnoreCase);
+
+            yield return (new CultureInfo("en-US"), CompareOptions.None);
+            yield return (new CultureInfo("en-US"), CompareOptions.IgnoreCase);
+            yield return (new CultureInfo("en-US"), CompareOptions.IgnoreSymbols);
+
+            yield return (CultureInfo.InvariantCulture, CompareOptions.None);
+            yield return (CultureInfo.InvariantCulture, CompareOptions.IgnoreCase);
+
+            // both en-US and Invariant cultures have an optimized execution path on Unix systems:
+            // https://github.com/dotnet/coreclr/blob/cd6bc26bdc4ac06fe2165b283eaf9fb5ff5293f4/src/System.Private.CoreLib/shared/System/Globalization/CompareInfo.Unix.cs#L35
+
+            // so we need one more culture to test the "slow path"
+            // Polish language has a lot of special characters, for example 'ch', 'rz', 'sz', 'cz' use two chars to express one ;)
+            // it also has a lot of characters with accent so we use it as an example of "complex" language
+            yield return (new CultureInfo("pl-PL"), CompareOptions.None);
+        }
+
+        [ParamsSource(nameof(GetCultureOptions))]
+        public (CultureInfo CultureInfo, CompareOptions CompareOptions) CultureOptions;
```

Comment:
Should this be a property (instead of a public field)?

Reply:
A property would be translated to a backing field with an inlinable setter and getter. I personally prefer to use fields in benchmarks (WYSIWYG).

Reply:
The same applies to `Count` (below). My suggestion was for consistency in the code.

Reply:
@gfoidl thanks, now I see the point. I've just pushed a commit to use fields for all `params`